lib/prelude.dx

```diff
@@ -35,14 +37,48 @@ def FToF32 (x : Float)   : Float32 = internalCast _ x
 def I64ToI (x : Int64)   : Int     = internalCast _ x
 def I32ToI (x : Int32)   : Int     = internalCast _ x
 def W8ToI  (x : Word8)   : Int     = internalCast _ x
+def W64ToI (x : Word64)  : Int     = internalCast _ x
 def IToI64 (x : Int)     : Int64   = internalCast _ x
 def IToI32 (x : Int)     : Int32   = internalCast _ x
 def IToW8  (x : Int)     : Word8   = internalCast _ x
+def IToW32 (x : Int)     : Word32  = internalCast _ x
+def W64ToW32 (x : Word64): Word32  = internalCast _ x
+def W32ToW64 (x : Word32): Word64  = internalCast _ x
+def IToW64 (x : Int)     : Word64  = internalCast _ x
 def IToF (x:Int) : Float = internalCast _ x
 def FToI (x:Float) : Int = internalCast _ x
 def I64ToRawPtr (x:Int64 ) : RawPtr = internalCast _ x
 def RawPtrToI64 (x:RawPtr) : Int64  = internalCast _ x
 
+'## Bitwise operations
+
+interface Bits a
+  (<<)  : a -> Int -> a
+  (>>)  : a -> Int -> a
+  (.|.) : a -> a -> a
+  (.&.) : a -> a -> a
+  (^)   : a -> a -> a
```

Comment:
I don't expect people to be writing too much bit manipulation in Dex so I kind of like the idea of wrapping them consistently with dots on both sides. Can you please do this for `^`, `<<` and `>>`?